### ISSUE TITLE
Incorrect query parsing error

### ISSUE THREAD

**Issue by Aurora-yzj** [neo4j/neo4j#13388]
Hello! I discovered a bug that Neo4j incorrectly parses my queries. The legitimate query was mistakenly reported as an incorrect query by Neo4j. The detailed queries are listed below. 
**Neo4j Version:** 5.14.0 Community
**Operating System:** Ubuntu 22.04
**API:** Docker

### Steps to reproduce
1. Pull the image: `docker pull neo4j`
2. Create a directory on the host that will be a mount for the data: `mkdir $HOME/neo4j-data`
3. Start a new container that mounts to this directory: `sudo docker run -d --name neo4j-test -p 17474:7474 -p 17687:7687  -v $HOME/neo4j-data:/data neo4j`
4. Go to the web interface at "http://127.0.0.1:17474", login and execute the following series of queries to build the database:
```
CREATE (n0 :L18{k110 : false, k111 : 934978338, id : 1, k107 : -1273360340, k109 : "ipzWgc"});
CREATE (n0 :L18:L1:L19{k11 : true, k10 : "61h6xCJdv", k7 : -2042171447, k8 : "aLncuRq", k9 : "XoR15", k110 : true, k112 : 853174432, k111 : -904716004, k114 : "pSEpK", k113 : 432224584, k116 : -1518965077, id : 2, k115 : false, k107 : -330887455, k118 : false, k117 : false, k109 : "kghCiRf", k108 : "ZquIs2rl", k119 : true});
CREATE (n0 :L18{k110 : false, k112 : 1846996640, k111 : 221999184, id : 3, k107 : 150988175, k109 : "GnCTdbdu", k108 : "FPkGrpH"});
CREATE (n0 :L18{k110 : false, k112 : -947100760, k111 : -1060256550, id : 4, k107 : 1705917083, k109 : "cfHwr6c", k108 : "FmwBI"});
CREATE (n0 :L18{k110 : false, k112 : -329365692, k111 : 853827871, id : 8, k107 : 1577916897, k109 : "B9jh3k", k108 : "nQmnE"});
CREATE (n0 :L18:L1{k110 : true, k112 : -1024647473, k111 : -75451858, k11 : false, id : 9, k10 : "caszHlXs5", k107 : -71081229, k7 : 1246284387, k8 : "tR4x5n8", k109 : "Fn8SL", k9 : "RfXbIiu", k108 : "dAtuR"});
CREATE (n0 :L18:L1{k110 : false, k112 : 395282026, k111 : 1372159616, k11 : true, id : 10, k10 : "1NaNO", k107 : -1223348445, k7 : -2024217477, k8 : "XVcrqq", k109 : "SpgJljRF8", k9 : "VRBf8FPPW", k108 : "qw6jSQbMA"});
CREATE (n0 :L18:L1{k110 : true, k112 : 466921996, k111 : 1266827294, k11 : false, id : 11, k10 : "yyPkMmUxV", k107 : -445931554, k7 : -697786717, k8 : "flOvYiA", k109 : "cwzWnC", k9 : "OsEtrUlTu", k108 : "Ed75l8"});
CREATE (n0 :L18:L1:L19{k11 : true, k10 : "7YfRcYcE8", k7 : -376054471, k8 : "Z0L2AVp", k9 : "zDkUnjsH", k110 : true, k112 : -989538447, k111 : -1249927471, k114 : "vhqzeFjEV", k113 : -1793222219, id : 12, k115 : false, k107 : -1259933720, k118 : true, k117 : false, k109 : "DIWwxhrb", k108 : "c6ZjT6XO", k119 : true});
CREATE (n0 :L18{k110 : false, k112 : 658999222, k111 : -1227926691, id : 13, k107 : 1150242752, k109 : "JtNKn10", k108 : "EQn3NZ2"});
MATCH (n0 {id : 3}), (n1 {id : 8}) MERGE(n0)-[r :T12{k222 : 879624812, k221 : -949840107, k224 : 1542675133, k223 : "CMxxu8", id : 16, k225 : true}]->(n1);
MATCH (n0 {id : 10}), (n1 {id : 9}) MERGE(n0)-[r :T12{k222 : 1802831735, k221 : -132617099, k224 : -442243796, k223 : "FO0kvq", id : 18, k225 : false}]->(n1);
MATCH (n0 {id : 12}), (n1 {id : 4}) MERGE(n0)-[r :T12{k222 : -1292958440, k221 : 1760787504, k224 : 737110382, k223 : "gDwDc", id : 19, k225 : false}]->(n1);
MATCH (n0 {id : 2}), (n1 {id : 9}) MERGE(n0)-[r :T12{k222 : 467766439, k221 : -2015298869, k224 : 27519868, k223 : "KwPg8MN4", id : 20, k225 : true}]->(n1);
MATCH (n0 {id : 8}), (n1 {id : 1}) MERGE(n0)-[r :T12{k222 : 2139060154, k221 : -39261920, k224 : -1646168314, k223 : "CmTcF", id : 21, k225 : false}]->(n1);
```
5. Run the query:
```
MATCH (n0 :L18)<-[r0 :T12]-(n1 :L18)<-[r1 :T12]-(n2 :L18) WHERE ((n2.k112) = 1846996640) OPTIONAL MATCH (n0 :L18)<-[r2 :T12]-(n1 :L18)<-[r3 :T12]-(n2), (n3 :L1 :L19)-[r4 :T12]->(n4), (n5 :L18 :L1)<-[r5 :T12]-(n6 :L18 :L1) WHERE (n1.k108) > ((n0.k109)+(n5.k9)) AND (n1.k108) = 'nQmnE' AND (n4.k111) = -1060256550 AND (n6.k107) = -1223348445 WITH (r3.k225) AS a0 WHERE (false OR a0) RETURN 1 AS a2
```

### Expected behavior
It is a legitimate query, so it should output the result `1`.

### Actual behavior
Neo4j reported this query as incorrect query, and output the error: `Don't know how to treat that as a predicate: String("nQmnE")`
![image](https://github.com/neo4j/neo4j/assets/24768784/9e64aa52-5b97-43b5-97a9-854d0af5e251)

To help you locate the issue, I found the `WITH` clause at the end of the query is related to reproducing this bug. 
Thanks for your help!

**Comment by htsedebenham** [comment]
Thanks for reporting :)

How are you using this query?  While I see the same error when using Docker, the same query gives the correct output when using Neo4j Desktop which might unblock you.

**Comment by Aurora-yzj** [comment]
> Thanks for reporting :)
> 
> How are you using this query? While I see the same error when using Docker, the same query gives the correct output when using Neo4j Desktop which might unblock you.

Thanks for the tip! But it is strange since this problem still occurs on my side when using Neo4j compiled from sources instead of Docker. 
![image](https://github.com/neo4j/neo4j/assets/24768784/5b375db2-7116-4240-b531-e9bee824536c)

**Comment by JonasHolgersson** [comment]
Duplicate of https://github.com/neo4j/neo4j/issues/13359
